image_cloudevents/modelmonitor.ipynb

```diff
@@ -0,0 +1,303 @@
+   "source": [
+    "## Enabling logs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "while True:\n",
+    "    DEPLOYMENT_ID = api.get_deployment_id(name=MONITOR_NAME)\n",
+    "    if DEPLOYMENT_ID:\n",
+    "        break\n",
+    "    print(\"waiting for deployment to come up\")\n",
+    "    time.sleep(api.wait_interval)\n",
+    "inference, INFERENCE_URL = wait_for_deployment_running(DEPLOYMENT_ID)\n",
+    "if not inference.enable_logs:\n",
+    "    print(\"Enabling logs\")\n",
+    "    serving = DkubeServing(user=DKUBEUSERNAME, name=MONITOR_NAME)\n",
+    "    serving.update_enable_logs(enable_logs=True)\n",
+    "    api.update_inference(serving)\n",
+    "print(\"Inference is up at URL: \", INFERENCE_URL)\n",
+    "%store INFERENCE_URL\n",
+    "%store DEPLOYMENT_ID"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initializing model monitor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm=DkubeModelmonitor(deployemnt_id = DEPLOYMENT_ID)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm.update_modelmonitor_basics(model_type=ModelType.Classification.value, \n",
+    "                               input_data_type=DataType.Image.value,\n",
+    "                               data_timezone=TimeZone.UTC.value)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm.update_image_data_shape(height=200, width=200, channel=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DKUBE_BASE_DATASET = \"chest-xray-data\"\n",
+    "training_data = f'{DKUBE_BASE_DATASET}:'+ DKUBEUSERNAME\n",
+    "train_data_version = 'v1:'+api.get_dataset_versions(\n",
+    "    DKUBEUSERNAME,\n",
+    "    DKUBE_BASE_DATASET)[0]['version']['uuid']\n",
+    "prediction_data = LIVE_DATASET + \":\"+ DKUBEUSERNAME\n",
+    "labelled_data = LIVE_DATASET + \":\"+ DKUBEUSERNAME\n",
+    "predict_data_format = \"cloudeventlogs\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adding train, predict, and labelled datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm.add_datasources(data_class=DatasetClass.Train.value,name=training_data,\n",
+    "                   version=train_data_version,\n",
+    "                   s3_subpath=DEPLOYMENT_ID)\n",
+    "mm.add_datasources(data_class=DatasetClass.Predict.value,name=prediction_data,\n",
+    "                   s3_subpath=DEPLOYMENT_ID, data_format=predict_data_format,\n",
+    "                   date_suffix=\"yyyy/mm/dd/hh\")\n",
```

Comment:
they don't know this yyyy/mm/dd/hh format for cloudevents generated by us. Can we make it optional and we can assume internally?

Reply:
we can ignore that, for cloudevents pipeline is using yyyy/mm/dd/hh only.